astropy/units/tests/test_physical.py

```diff
@@ -477,3 +478,34 @@ def teardown_method(self):
                 f"the physical type for {unit}, which was added for"
                 f"testing, was not deleted."
             )
+
+
+@pytest.mark.parametrize(
+    "method, expected",
+    [
+        ("title", 'Length'),
+        ("isalpha", True),
+        ("isnumeric", False),
+        ("upper", 'LENGTH')
+    ],
+)
+def test_that_str_methods_work_with_physical_types(method, expected):
+    """
+    Test that str methods work for `PhysicalType` instances while issuing
+    a deprecation warning.
+    """
+    with pytest.warns(AstropyDeprecationWarning):
```

Comment:
At least do a partial match to make sure the deprecation warning is actually the one you want.
```suggestion
    with pytest.warns(AstropyDeprecationWarning, match="PhysicalType instances"):
```